docker-app/qfieldcloud/core/admin.py

```diff
@@ -71,6 +75,124 @@ def admin_urlname_by_obj(value, arg):
+            """
+            WITH u AS (
+                SELECT
+                    DISTINCT ON (COALESCE(ae.email, u.email)) COALESCE(ae.email, u.email) AS "email",
+                    u.id,
+                    u.username,
+                    u.first_name,
+                    u.last_name,
+                    u.type,
+                    u.date_joined
+                FROM
+                    core_user u
+                    LEFT JOIN account_emailaddress ae ON ae.user_id = u.id AND ae.primary
+                WHERE
+                    COALESCE(ae.email, u.email) IS NOT NULL
+                    AND COALESCE(ae.email, u.email) != ''
+                ORDER BY
+                    COALESCE(ae.email, u.email),
+                    u.type
+            )
+            SELECT
+                u.id,
+                u.username,
+                u.email,
+                u.first_name,
+                u.last_name,
+                u.date_joined,
+                u.type,
+                oo.id AS "owner_id",
+                oo.username AS "owner_username",
+                oo.email AS "owner_email",
+                oo.first_name AS "owner_first_name",
+                oo.last_name AS "owner_last_name",
+                oo.date_joined AS "owner_date_joined"
+            FROM
+                u
+                LEFT JOIN account_emailaddress ae ON ae.user_id = u.id
+                LEFT JOIN core_organization o ON o.user_ptr_id = u.id
+                LEFT JOIN u oo ON oo.id = o.organization_owner_id
+            ORDER BY u.id
+            """
+        )
+        return (
+            UserEmailDetails(
+                row.id,
+                row.username,
+                row.first_name,
+                row.last_name,
+                row.type,
+                row.email,
+                row.date_joined,
+                row.owner_id,
+                row.owner_username,
+                row.owner_email,
+                row.owner_first_name,
+                row.owner_last_name,
+                row.owner_date_joined,
+            )
+            for row in raw_queryset
+        )
+
+    @admin.action(description="Export all users' email contact details to .csv")
+    def export_emails_to_csv(self, request) -> StreamingHttpResponse:
+        """ "Export all users' email contact details to .csv"""
+
+        class PseudoBuffer:
+            # Good idea from https://docs.djangoproject.com/en/4.1/howto/outputting-csv/
+            def write(self, value):
+                return value
+
+        email_rows = self.gen_users_email_addresses()
+        pseudo_buffer = PseudoBuffer()
+        writer = csv.writer(pseudo_buffer)
+        writer.writerow(col_name for col_name in UserEmailDetails._fields)
+
+        return StreamingHttpResponse(
+            (writer.writerow(row) for row in email_rows),
+            content_type="text/csv",
+            headers={
+                "Content-Disposition": 'attachment; filename="exported_users_emails.csv"'
```

Comment:
Can you please add the current datetime in the filename? Something like `qfc_user_email_20230227_120900.csv`.